python/ray/autoscaler/aws/node_provider.py

```diff
@@ -204,15 +208,30 @@ def create_node(self, node_config, tags, count):
         # SubnetIds is not a real config key: we must resolve to a
         # single SubnetId before invoking the AWS API.
         subnet_ids = conf.pop("SubnetIds")
-        subnet_id = subnet_ids[self.subnet_idx % len(subnet_ids)]
-        self.subnet_idx += 1
-        conf.update({
-            "MinCount": 1,
-            "MaxCount": count,
-            "SubnetId": subnet_id,
-            "TagSpecifications": tag_specs
-        })
-        self.ec2.create_instances(**conf)
+
+        max_retries = 5
+        for attempt in range(1, max_retries + 1):
```

Comment:
Why is this needed?

Reply:
This isn't directly related to the PR but occasionally I'll get an issue with the subnet not being compatible with `Security group X and subnet Y belong to different networks`. Since we randomly choose a subnet, I just made it retry a couple times.